### ISSUE TITLE
Unwanted image rotation on newer Androids after compression

### ISSUE THREAD

**Issue by MrTurtlev2** [numandev1/react-native-compressor#280]
Android >= 12
ReactNative: 0.68.0
react-native-compressor: 1.8.24 

After compressing the image manually, it rotates by 90 degrees. However, the automatic compression method works correctly. This issue was also present on iOS, but it was resolved with version 1.8.24.

**Comment by MrTurtlev2** [comment]
I found temporary solution for this issue.

go to ImageCompressor.ks located in:
node-modues/react-native-compressor/android/scr/main/java/com/reactnativecompressor/Image

then replace current function "manualCompressImage" with this code:

```
fun manualCompressImage(imagePath: String?, options: ImageCompressorOptions, reactContext: ReactApplicationContext?): String? {
    val image = if (options.input === ImageCompressorOptions.InputType.base64) decodeImage(imagePath) else loadImage(imagePath)
    val resizedImage = resize(image, options.maxWidth, options.maxHeight)
    val isBase64 = options.returnableOutputType === ImageCompressorOptions.ReturnableOutputType.base64
    val uri = Uri.parse(imagePath)
    val imagePathNew = uri.path
    var scaledBitmap: Bitmap? = correctImageOrientation(resizedImage, imagePathNew)
    val imageDataByteArrayOutputStream = compress(scaledBitmap, options.output, options.quality, options.disablePngTransparency)
    val compressedImagePath = encodeImage(imageDataByteArrayOutputStream, isBase64, options.output.toString(), imagePath, reactContext)
    if (isCompressedSizeLessThanActualFile(imagePath!!, compressedImagePath)) {
        return compressedImagePath
    } else {
        MediaCache.deleteFile(compressedImagePath!!)
        return slashifyFilePath(imagePath)
    }
} 
```

after updating the code just patch package with patch-package

`yarn patch-package react-native-compressor`

**Comment by MatteoGauthier** [comment]
I still have the issue with image with "face" camera device

**Comment by MayThinnCho** [comment]
Have you managed to solve this issue? I'm facing the same problem.

**Comment by Equilibriumty** [comment]
I had the same issue when taking a photo using front camera on Android 14. Found the solution - replace `correctImageOrientation` with this implementation:

```
fun correctImageOrientation(bitmap: Bitmap?, imagePath: String?): Bitmap? {
    if (bitmap == null || imagePath == null) return bitmap

    return try {
        val exif = ExifInterface(imagePath)
        val orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
        val matrix = Matrix()

        when (orientation) {
            ExifInterface.ORIENTATION_NORMAL -> return bitmap
            ExifInterface.ORIENTATION_FLIP_HORIZONTAL -> {
                matrix.setScale(-1f, 1f)
            }
            ExifInterface.ORIENTATION_ROTATE_180 -> {
                matrix.setRotate(180f)
            }
            ExifInterface.ORIENTATION_FLIP_VERTICAL -> {
                matrix.setScale(1f, -1f)
            }
            ExifInterface.ORIENTATION_TRANSPOSE -> {
                matrix.setRotate(90f)
                matrix.postScale(-1f, 1f)
            }
            ExifInterface.ORIENTATION_ROTATE_90 -> {
                matrix.setRotate(90f)
            }
            ExifInterface.ORIENTATION_TRANSVERSE -> {
                matrix.setRotate(-90f)
                matrix.postScale(-1f, 1f)
            }
            ExifInterface.ORIENTATION_ROTATE_270 -> {
                matrix.setRotate(-90f)
            }
            else -> return bitmap
        }

        Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
    } catch (e: IOException) {
        e.printStackTrace()
        bitmap
    }
}
```

**Comment by Equilibriumty** [comment]
> > I had the same issue when taking a photo using front camera on Android 14 with both `compressionMode: 'auto'` and `compressionMode: 'manual'`. Fix from the comment above didn't help in my case. Found the solution - replace `correctImageOrientation` with this implementation:
> > fun correctImageOrientation(bitmap: Bitmap?, imagePath: String?): Bitmap? {
> > if (bitmap == null || imagePath == null) return bitmap
> > ```
> > return try {
> >     val exif = ExifInterface(imagePath)
> >     val orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
> >     val matrix = Matrix()
> > 
> >     when (orientation) {
> >         ExifInterface.ORIENTATION_NORMAL -> return bitmap
> >         ExifInterface.ORIENTATION_FLIP_HORIZONTAL -> {
> >             matrix.setScale(-1f, 1f)
> >         }
> >         ExifInterface.ORIENTATION_ROTATE_180 -> {
> >             matrix.setRotate(180f)
> >         }
> >         ExifInterface.ORIENTATION_FLIP_VERTICAL -> {
> >             matrix.setScale(1f, -1f)
> >         }
> >         ExifInterface.ORIENTATION_TRANSPOSE -> {
> >             matrix.setRotate(90f)
> >             matrix.postScale(-1f, 1f)
> >         }
> >         ExifInterface.ORIENTATION_ROTATE_90 -> {
> >             matrix.setRotate(90f)
> >         }
> >         ExifInterface.ORIENTATION_TRANSVERSE -> {
> >             matrix.setRotate(-90f)
> >             matrix.postScale(-1f, 1f)
> >         }
> >         ExifInterface.ORIENTATION_ROTATE_270 -> {
> >             matrix.setRotate(-90f)
> >         }
> >         else -> return bitmap
> >     }
> > 
> >     Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
> > } catch (e: IOException) {
> >     e.printStackTrace()
> >     bitmap
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > }
> > And then run `yarn patch-package react-native-compressor` Double check that patch has been applied via `yarn`
> > Rebuild the app
> 
> This Function is only used in the autoCompression Mode and if you are using Manual [@MrTurtlev2](https://github.com/MrTurtlev2) Code needs to be Added in the Manual Function so it can Correct the Orientation.

Yes, thanks for correcting me. I've updated my comment accordingly

**Comment by TiTnOuK** [comment]
I have same problem on on Android devices

Android >= 10
react-native: 0.80.2
react-native-compressor: 1.12.0

**Comment by XChikuX** [comment]
@numandev1 Any reason we are not merging these changes?

The above solutions provided by the community work great..

**Comment by Rengod95** [comment]
@numandev1 Why don't respond to the issue? I think that is correct solution.
